### PR TITLE
Case list explorer uat feedback

### PR DESCRIPTION
**Pull request by proteusvacuum**
Now with syntax highlighting! 
I am adding a new JS dependency (ace editor) so that in the future I have more control over the autocomplete. For free this gives us syntax highlighting, and I imagine this (+ autocomplete) would be really fun to use in other places where we want you to write XPath queries. 

screenshot from 2018-06-20 08-46-23

:blowfish: / :blowfish:

### CODE REVIEW

**Comment by orangejenny** [comment]
Screenshot looks great

File: corehq/apps/case_search/filter_dsl.py [hunk 1]
```diff
@@ -204,6 +209,39 @@ def visit(node):
     return visit(node)
 
 
+def build_filter_from_xpath(domain, xpath):
+    try:
```
Comment (orangejenny) [inline corehq/apps/case_search/filter_dsl.py]: The except clauses look like they could be DRYer.

File: corehq/apps/reports/static/reports/js/filters/main.js [hunk 2]
```diff
@@ -150,15 +150,19 @@ hqDefine("reports/js/filters/main", [
...
                 data = $el.data();
             var model = caseListExplorer.casePropertyColumns(data.initialvalue, data.columnsuggestions);
             $el.koApplyBindings(model);
         });
         $casePropertyColumns.on('keyup', function(){
-            $('#fieldset_explorer_columns').trigger('change');
+            $fieldsetExplorerColumns.trigger('change');
+        });
+        $fieldsetExplorerColumns.on('hidden.bs.collapse show.bs.collapse', function(e){
+            e.stopPropagation();
```
Comment (orangejenny) [inline corehq/apps/reports/static/reports/js/filters/main.js]: Might be nice to comment the `stopPropagation` call.

File: corehq/apps/reports/templates/reports/filters/explorer_columns.html [hunk 3]
```diff
@@ -2,57 +2,62 @@
...
-                        <span class="pull-right" id="panel-chevron">
+                <h4 class="panel-title">
+                    <a role="button" data-toggle="collapse" data-parent="#case-list-explorer-columns" href="#columns-editor">
+                        <span id="panel-chevron">
                             <i class="fa fa-chevron-down"></i>
                         </span>
-                    </h4>
-                </a>
+                        {% trans "Edit Columns" %}
+                        <small>(</small><span data-bind="foreach: properties"><!--ko if: $index() != 0 && name() !=='' -->, <!--/ko--><small data-bind="text: name"></small></span><small>)</small>
```
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/filters/explorer_columns.html]: Why would a property's name be blank?

Comment (proteusvacuum) [reply corehq/apps/reports/templates/reports/filters/explorer_columns.html]: when you click the `+ Add column` button, the new property gets created without a name, so it adds an awkward comma at the end of the list.

File: corehq/apps/reports/static/reports/js/filters/case_list_explorer_knockout_bindings.js [hunk 4]
```diff
@@ -1,42 +1,44 @@
-hqDefine("reports/js/filters/case_list_explorer_knockout_bindings", ['jquery', 'knockout', 'hqwebapp/js/atwho'], function($, ko, atwho) {
+hqDefine("reports/js/filters/case_list_explorer_knockout_bindings", ['jquery', 'knockout', 'hqwebapp/js/atwho', 'ace-builds/src-min-noconflict/ace'], function($, ko, atwho) {
```
Comment (orangejenny) [inline corehq/apps/reports/static/reports/js/filters/case_list_explorer_knockout_bindings.js]: Curious why you picked this syntax highlighter in particular and how it compares to the two we already use (CodeMirror and SyntaxHighlighter). Eventually, I'd like to standardize on one.

**Review by orangejenny — APPROVED** [review]

**Review by gbova — APPROVED** [review]

**Pull request merged**